cmd/format-config-v1.go

```diff
@@ -793,8 +794,8 @@ func checkFormatXLValues(formatConfigs []*formatConfigV1) (int, error) {
 			return i, err
 		}
 		if len(formatConfigs) != len(formatXL.XL.JBOD) {
-			return i, fmt.Errorf("Number of disks %d did not match the backend format %d",
-				len(formatConfigs), len(formatXL.XL.JBOD))
+			return i, fmt.Errorf("%s disk is already being used in another erasure deployment. (Number of disks specified: %d but the number of disks found in the %s disk's format.json: %d.)",
```

Comment:
This we should be perhaps done in @kannappanr @ebozduman logging changes, can this be done there instead.

Reply:
@harshavardhana 
I am not sure if that would be the right approach. 
As part of the logging changes, we are not going to  reword the error messages. We'll use the existing error messages and wording in a different structure. 
So, I think it is better to address this issue here instead of @kannappanr or me looking for the right wording for this specific error condition as part of the logging change effort. 
My 2 cents!

Reply:
I think we were planning to move messages into a single file, this fix is not needed right now AFAICS. Once there is a format review/change of all messages we should take this up. There is no real urgency in i can see we need to make this change. Perhaps we can do something simpler than a message like this.

```
1st disk is already being used in another erasure deployment. (Number of disks specified: 8 but the number of disks found in the 1st disk's format.json: 4.) 
```

If our intention is to give an actionable message such as `cleanup`, `re-format` etc. `format.json` shouldn't be exposed in the message as this doesn't add anything actionable to the user. Neither we write about `format.json` 

Current message gives a whole picture `Number of disks did not match the backend format` , but it is still too low level which can be changed to 

`Number of disks specified %d do not match existing erasure deployment %d`

Reply:
Also i think logging changes such as messages should be a single PR which changes a whole sub-system of messages, like fixes for all messages under `xl-v1.go`

Reply:
This PR was sent as #5239 was assigned to me, if it should be fixed in an umbrella of log fixes, sure. Anything is fine.

Reply:
Yeah my only thing was duplication of effort, since there might be many more such messages which might need improvement - they all can come with a formal review. This way we don't add more custom messages. 

We can take this PR in but we should talk to AB about changing error messages at this point in time.

Reply:
When Kannappan and I talked to AB about the logging project, we've also discussed the need for rewording/simplifying server error messages.
The decision is that there will a separate effort/issue for it, since we'll have to touch almost every error message and reword them, and strip off `%s` formatting, etc. when we introduce error codes for each and every error message.

Reply:
Thanks @ebozduman this was my line of thought as well.

Reply:
@harshavardhana The change introduced in this PR does not deal with the error message itself. This PR changes the `cause` of the error. Server logging is not dealing with rewording `cause` messages, atleast not right now. I believe we can review this PR.